src/layout/ThemeContext.tsx

```diff
@@ -8,38 +8,47 @@ import { useLocalStorage } from 'src/locale/useLocalStorage'
 
 export interface ThemeContextInterface {
   toggleTheme: () => void
+  toggleLanguage: () => void
   isDarkTheme?: boolean
 }
 
 const ThemeContext = createContext({} as ThemeContextInterface)
 
-const darkTheme = createTheme({
-  palette: {
-    mode: 'dark',
-  },
-})
-
-const lightTheme = createTheme({
-  palette: {
-    mode: 'light',
-  },
-})
-
 const { defaultAlgorithm, darkAlgorithm } = theme
 
 export const ThemeProvider: FC<PropsWithChildren> = ({ children }) => {
   const [isDarkTheme, setIsDarkTheme] = useLocalStorage<boolean>('isDarkTheme')
-
+  const [direction, setDirection] = useState<'ltr' | 'rtl'>('rtl')
```

Comment:
do we need a state variable here? sounds like we can use derived state here

Reply:
When I checked previously with a useEffect on the i18n.language to see where it is changing and re-create the theme.direction based on it, it didn't catch those changes so I created this state to update when changing language. I'll test it again and remove it if i can successfully derive the direction from the state.

Reply:
I think it can be calculated without hooks at all - it's a primitive string so there's no harm in recalculating it without memoization (unlike object references that could trigger unnecessary renders )

Reply:
You were right. I fixed this in both branches by mistake. Is that still ok to merge them now?